api-call.py

```diff
@@ -1,29 +1,226 @@
+                             data=payload,
+                             headers=headers,
+                             timeout=10)
+
+    if response.status_code == 200:
+        # Request was successful
+        access_token = response.json()['access_token']
+        if access_token:
+            access_token = enhancer + access_token
+        return access_token
+    else:
+        log_error(str(http_failed) + str(response.status_code))
 
 
 def get_api_data(url, parameters, header):
-    response = requests.get(f"{url}", params=parameters, headers=header)
+    """Get Request for getting the data from API
+
+    :param url: URL of the API
+    :param parameters: The parameters for API
+    :param header: The headers for API with authorization
+    """
+
+    response = requests.get(f"{url}",
+                            params=parameters,
+                            headers=header,
+                            timeout=10)
     if response.status_code == 200:
-        print("sucessfully fetched the data with parameters")
-        format_result(response.json())
+        # format_result(response.json())
+        return response.json()
     else:
-        print(
-            f"There's a {response.status_code} error with this request")
+        print(response.status_code)
+        generated_error = 'There is a '
+        + str(response.status_code)
+        + ' error with this request'
+        log_error(generated_error)
+
 
 def format_result(data):
-    text = json.dumps(data, sort_keys=True, indent=4)
+    """Prints data in a specified format
+
+    :param data: data fetched from API
+    """
+
+    text = json.dumps(data, indent=4)
     print(text)
 
 
+def show_tasks():
+    """Show tasks to choose as a Menu
+    """
+
+    config = load_from_config()
+    # Get the API options from the config
+    api_options = config['apiOptions']
+    print(config['generalStrings']['whichApi'])
+    for i, option in enumerate(api_options, start=0):
+        print(f"{i}. {option}")
+
+
+def get_user_choice():
+    """Get user input from menu
+
+    :returns: The user choice
+    """
+
+    config = load_from_config()
+    return input(config['generalStrings']['enterChoice'])
+
+
+def get_url(choice):
+    """Get URL for API
+
+    :param choice: user choice input
+    :returns: API URL Object after user choice
+    """
+
+    config = load_from_config()
+    access_token = get_access_token()
+    content_type = config['accessToken']['headers']['Content-Type']
+    header = {'Content-Type': content_type, 'Authorization': access_token}
+    url_obj = {}
+    if choice == '0':
+        url_obj = {'url': config['generalStrings']['exiting'],
+                   'parameters': {},
+                   'header': header}
+    elif choice == '1':
+        url_obj = {'url': config['apiUrls']['beaverBus'],
+                   'parameters': {},
+                   'header': header}
+    elif choice == '2':
+        url_obj = {'url': config['apiUrls']['terms'],
```

Comment:
It's a small thing, but you probably have noticed that you actually have to load the config every time when this function is called, and some fixed strings are also reloaded too (e.g. `config['apiUrls']['beaverBus']`). Theoretically, this stuff should only be loaded once (when the script starts). How would you improve that? Hint: OOP.

Reply:
Even I was thinking about these multiple load statements. Thanks.

Reply:
Converted in Python class.